insonmnia/hub/router_linux.go

```diff
@@ -37,7 +35,7 @@ func (r *ipvsRouter) RegisterRoute(ID string, protocol string, realIP string, re
 		return nil, err
 	}
 
-	serviceOptions, err := gateway.NewServiceOptions(host.String(), port, protocol)
+	serviceOptions, err := gateway.NewServiceOptions(host.String(), port, protocol.String())
```

Comment:
Maybe it's better to change NewServiceOptions signature to accept a protocol as Protocol type instead of string?

Reply:
Actually introducing `Protocol` was really bad idea, because it gives no proc except type safety, while giving unnecessary complexity.
I'll revert these back.

Reply:
Also I felt it out of the scope.